api/src/test/java/org/openmrs/module/cohort/api/dao/CohortTypeGenericDaoTest.java

```diff
@@ -79,4 +82,27 @@ public void shouldVoidCohortType() {
 		assertThat(voidedCohortType, nullValue());
 	}
 	
+	@Test
+	public void shouldPurgeCohortType() {
+		CohortType cohortType = dao.get(COHORT_TYPE_UUID);
+		assertNotNull(cohortType);
+		dao.delete(cohortType);
+		CohortType afterAttemptPurge = dao.get(COHORT_TYPE_UUID);
+		assertNull(afterAttemptPurge);
+	}
+	
+	@Test
+	public void shouldThrowExceptionForPurgeCohortType() {
+		CohortType cohortType1 = dao.get("94517bf9-d9d6-4726-b4f1-a2dff6b36e2d");
+		assertNotNull(cohortType1);
+		
+		try {
+			dao.delete(cohortType1);
+		}
+		catch (Exception e) {
```

Comment:
Can we instead use `@Test(expected = ...`?

Reply:
@dkayiwa I have updated the unit test as follows:

@Test(expected = Exception.class)
public void shouldThrowExceptionForPurgeCohortType() {
    CohortType cohortType = dao.get("94517bf9-d9d6-4726-b4f1-a2dff6b36e2d");
    assertNotNull(cohortType);
    dao.delete(cohortType);
}

However, in this test, the line dao.delete(cohortType) does not throw any exception (when I tested it on Postman, it threw an error when the cohort type was associated with a cohort). Interestingly, if I call something like dao.get(UUID) after dao.delete(), then the expected exception is thrown during the dao.delete() operation.

Reply:
How about doing `Context.flushSession()` after dao.delete?

Reply:
@dkayiwa  It is working, changes are pushed you can review.